src/subsplit_dag.cpp

```diff
@@ -21,53 +25,54 @@ SubsplitDAG::SubsplitDAG(size_t taxon_count,
          "Taxon count mismatch in SubsplitDAG constructor.");
   auto [edge_indexer, index_to_child, rootsplits] =
       ProcessTopologyCounter(topology_counter);
+  BuildTaxonMap(tag_taxon_map);
   BuildNodes(index_to_child, rootsplits);
   BuildEdges(index_to_child);
   BuildDAGEdgesFromEdgeIndexer(edge_indexer);
   AddLeafSubsplitsToDAGEdgesAndParentToRange();
   CountTopologies();
-}
 
-SubsplitDAG::SubsplitDAG(const RootedTreeCollection &tree_collection)
-    : SubsplitDAG(tree_collection.TaxonCount(), tree_collection.TopologyCounter(),
-                  tree_collection.TagTaxonMap()) {}
+  Assert(IsValidTaxonMap(),
+         "SubsplitDAG::SubsplitDAG(): Given Taxon Map does not represent a valid map. "
+         "Taxon ids do not cover all indices from 0 to taxon_count.");
+}
 
-// ** Comparator methods:
+// ** Comparator
 
 int SubsplitDAG::Compare(const SubsplitDAG &lhs, const SubsplitDAG &rhs) {
   // (1) Compare Taxon Sizes.
-  if (lhs.TaxonCount() != rhs.TaxonCount()) {
+  int taxon_diff = lhs.TaxonCount() - rhs.TaxonCount();
+  if (taxon_diff != 0) {
     // #350 let's talk about -100 vs -200 here.
-    return -100;
+    return taxon_diff;
   }
   // Create translation map (lhs->rhs) for bitset clades.
   auto taxon_map = SubsplitDAG::BuildTaxonTranslationMap(lhs, rhs);
   // (2) Compare Subsplit Nodes.
-  auto lhs_nodes = lhs.GetSortedVectorOfNodeBitsets();
-  auto rhs_nodes = rhs.GetSortedVectorOfNodeBitsets();
+  auto nodes_a = lhs.GetSortedVectorOfNodeBitsets();
```

Comment:
I'm not sure why this is an improvement. We have arguments lhs and rhs, and now we are changing to a and b.